### ISSUE TITLE
VCheckbox element logical state not updated to match the state stored in the application

### ISSUE THREAD

**Issue by sarayourfriend** [WordPress/openverse#4087]
## Description

`VCheckbox`, represented as a toggle rather than a checkbox, is broken for users who rely on screen readers. 

## Reproduction

1.  Visit https://openverse.org/privacy
2.  Scroll to the end of the page, and inspect the "Record custom events and page views for analytics." input, in particular the `input type="checkbox"` element that backs the visual toggle.
3.  Identify whether the input has a `checked` attribute, and the value of the attribute. If the input is toggled on, this should be `checked="checked"`. If it is off, the input will not have a `checked` attribute.
4. Toggle the input by clicking on the element. Inspect the element again, and note that the `checked` attribute has not changed from the previous step. If it was not present, it will still not be present. If it was present, it will still be `checked="checked"`.

**This does _not_ affect the actual stored value of the checkbox, i.e., the value the application references**. You can confirm this in two ways: first, by reloading the page and seeing that the checkbox will be in the state you toggled it to (make sure to toggle to the opposite state for this to be meaningful, of course); second, by inspecting the relevant cookie stored by the application using your browser's developer tools. The value will match the visual state of the toggle, despite the checkbox itself not matching.

## Additional context

Found while reviewing https://github.com/WordPress/openverse/pull/4084 and debugging one of the new Playwright tests added in that PR, which cannot work due to the component's failure to reflect the toggle's state in the semantic HTML.

Marked critical because we have two live features that depend on this, both of which must be communicated about in absolute and clear terms: the analytics toggle, and the sensitive content search (and blur) toggles.

**Comment by sarayourfriend** [comment]
Apparently I am wrong: this is intentional and not a bug in our application: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#checked

> A [boolean](https://developer.mozilla.org/en-US/docs/Glossary/Boolean/HTML) attribute indicating whether this checkbox is checked by default (when the page loads). It does not indicate whether this checkbox is currently checked: if the checkbox's state is changed, this content attribute does not reflect the change.

All the more reason why in https://github.com/WordPress/openverse/pull/4084 we _must_ use semantic checks (e.g., `toBeChecked()`), rather than explicitly checking attribute values.

**Comment by obulat** [comment]
This problem in prod exists in the Switchable features and Non switchable features sections only. If you try to check the top-most Analytics switch, it works fine.

Whenever we have some problem with rendering or hydration when Vue cannot append a child node, we get the behavior you currently see in prod. The `checked` state is changed, VoiceOver on Mac states that the feature is checked/unchecked, but the app value for the feature stays the same. If you try to click on one of the switchable features and refresh, you'll see that the value did not update.

**Comment by sarayourfriend** [comment]
@obulat is your comment meant for a different issue? This was for an accessibility bug I thought affected all checkboxes (it doesn't, but that's what was going on). Did you mean to leave the comment on https://github.com/WordPress/openverse/issues/4082?

**Comment by obulat** [comment]
@sarayourfriend, right, I didn't fully understand that this only refers to the accessibility issue.
